keps/sig-auth/0034-20190123-harden-default-discovery-bindings.md

```diff
@@ -0,0 +1,71 @@
+---
+kep-number: 34
+title: Harden Default RBAC Discovery ClusterRole(Binding)s
+authors:
+  - "@dekkagaijin"
+owning-sig: sig-auth
+participating-sigs:
+  - sig-auth
+  - sig-api-machinery
+reviewers:
+  - "@liggitt"
+  - "@tallclair"
+approvers:
+  - "@liggitt"
+  - "@tallclair"
+editor: TBD
+creation-date: 2019-01-23
+last-updated: 2019-01-23
+status: provisional
+---
+
+# Harden Default RBAC Discovery ClusterRole(Binding)s
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a KEP and for highlighting any additional information provided beyond the standard KEP template.
+[Tools for generating][] a table of contents from markdown are available.
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Implementation History](#implementation-history)
+
+## Summary
+
+The aim of this change is to remove the `system:unauthenticated` subjects from the `system:discovery` and `system:basic-user` ClusterRoleBindings, while preserving unauthenticated access to a genuinely non-sensitive subset of the current APIs (e.g. `GET /healthz`, `GET /version`). This will improve the default privacy and security posture for new clusters.
```

Comment:
> while preserving unauthenticated access to a genuinely non-sensitive subset of the current APIs (e.g. `GET /healthz`, `GET /version`)

What makes these APIs less sensitive? In particular, version information could be used by scanner bots looking for unpatched clusters.

Reply:
Sure, but the alternative would probably be to try exploit(s) and hope for the best.

`/version` is called out as being non-sensitive, and is called by `kubectl version`: https://github.com/kubernetes/kubernetes/issues/45366